mqlc/builtin_map.go

```diff
@@ -203,6 +203,67 @@ func compileDictContainsOnly(c *compiler, typ types.Type, ref uint64, id string,
 	return types.Bool, nil
 }
 
+func compileDictContainsNone(c *compiler, typ types.Type, ref uint64, id string, call *parser.Call) (types.Type, error) {
+	if call == nil || len(call.Function) != 1 {
+		return types.Nil, errors.New("function " + id + " needs one argument (dict)")
+	}
+
+	f := call.Function[0]
+	if f.Value == nil || f.Value.Operand == nil {
+		return types.Nil, errors.New("function " + id + " needs one argument (dict)")
+	}
+
+	val, err := c.compileOperand(f.Value.Operand)
+	if err != nil {
+		return types.Nil, err
+	}
+
+	valType, err := c.dereferenceType(val)
+	if err != nil {
+		return types.Nil, err
+	}
+
+	chunkId := "==" + string(typ)
+	if valType != typ {
+		chunkId = "==" + string(valType)
+		_, err := llx.BuiltinFunctionV2(typ, chunkId)
+		if err != nil {
+			return types.Nil, errors.New("called '" + id + "' with wrong type; either provide a type " + typ.Label() + " value or write it as an expression (e.g. \"_ == 123\")")
+		}
+	}
+
+	// .containsNone
+	c.addChunk(&llx.Chunk{
+		Call: llx.Chunk_FUNCTION,
+		Id:   "containsNone",
```

Comment:
When do we set a `$` like in `$all` and when not?

Reply:
Generally when we don't want the given method to be exposed. It signifies internal methods, often helper methods to support other methods